### PR TITLE
Adding polymorphic relationship

### PR DESCRIPTION
**Pull request by orhanna14**
With new model, 'Picture'. This currently has a failing spec and is a WIP. I have it as far as visiting a route (that may not exist yet). 

I would like to know, what kind of tests should I include on this relationship? Should I do more planning before I code it out?

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -13,14 +13,19 @@ def new
...
+    @recipe = @user.recipes.find(params[:id])
+  end
+
   def create
     @recipe = Recipe.new(recipe_params)
     @recipe.user = User.find(params[:user_id])
     if @recipe.save
-      flash[:success] = t("success")
-      redirect_to "/"
+      flash[:success] = I18n.t('layouts.application.success')
```
Comment (edwardloveall) [inline app/controllers/recipes_controller.rb]: This is awesome that you're doing localization so early on in the process. One thing that stuck out to me is that this success message for creating a recipe is directly under the `application` key. That feels odd to me since this is an recipe not an _application_ being created.

How would you feel about instead changing this string to be under `recipes.create.success`? Bonus if you do that, you can shorten this to `t(".success")` since rails will know you're in the `create` action of the `recipes` controller.

Note: I had to look this up to remember where things should be placed and I found a perfect example [here](https://guides.rubyonrails.org/i18n.html#lazy-lookup) with the BooksController.

File: spec/system/user_adds_picture_to_recipe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    user = create(:user)
+    recipe = create(:recipe, user: user)
+
+    sign_in_with(user.email, user.password)
+    visit '/'
+    click_link I18n.t('layouts.application.view_recipes')
+    click_link I18n.t('recipes.index.show_recipe')
+    click_button I18n.t('recipes.show.add_picture')
+
+    expect(page).to have_current_path(recipe_pictures_path(recipe.id))
```
Comment (edwardloveall) [inline spec/system/user_adds_picture_to_recipe_spec.rb]: I think this is a good test but I think an even better test would be one that gets all the way through to adding an image. I haven't ever done this (I've only done it with paperclip) so I'm not sure exactly the process. If you're up for reading a bunch, I found these these things:

* https://www.neontsunami.com/posts/testing-activestorage-uploads-in-rails-52
* https://collectiveidea.com/blog/archives/2017/01/16/testing-an-uploaded-file-in-rspec

So some mixture of configuring active storage for a testing environment (which may already be done because this is rails 6) and then reading and uploading a file in the test. There's a lot of detail I'm skipping over here but I don't yet know the answer.

I'd be happy to pair with you on a Friday sometime if you like! Alternatively you can ask questions here if you get stuck.

Comment (orhanna14) [reply spec/system/user_adds_picture_to_recipe_spec.rb]: Thanks Edward! I'll do some reading and try to implement this on my own, and follow up with you for that offer :)
